geniza/corpus/management/commands/googledoc_to_annotation.py

```diff
@@ -0,0 +1,389 @@
+            row_start = (i * 2) + 2
+            # extract the annotation block label (first column)
+            block_label = tds[row_start].get_text()
+            # extract the translation (second column)
+            translation_cell = tds[row_start + 1]
+            # first, process and cleanup all the tags
+            for tag in translation_cell.descendants:
+                try:
+                    # remove unneeded attributes
+                    tag.attrs = {
+                        key: value
+                        for (key, value) in tag.attrs.items()
+                        if key not in self.REMOVE_ATTRIBUTES
+                    }
+                    # remove unnecessary spans that Google Docs adds to each li
+                    for span in tag.find_all("span"):
+                        span.unwrap()
+                    # clear out any unneeded newlines exposed by the previous step
+                    tag.smooth()
+                except AttributeError:
+                    # 'NavigableString' object has no attribute 'attrs'; fine to ignore
+                    pass
+            html = "\n".join([str(tag) for tag in translation_cell.contents])
+
+            # create an annotation
+            anno = self.new_translation_annotation()
+
+            # place on the first canvas
+            anno.target.source.id = iiif_canvas
+            # apply to the full canvas using % notation
+            # (using nearly full canvas to make it easier to edit zones)
+            anno.target.selector.value = "xywh=percent:1,1,98,98"
+
+            # add html and optional label to annotation text body
+            if not unicodedata.is_normalized("NFC", html):
+                self.normalized_unicode.add(pgpid)
+                html = unicodedata.normalize("NFC", html)
+            anno.body[0].value = html
+
+            # check if label text requires normalization
+            if not unicodedata.is_normalized("NFC", block_label):
+                self.normalized_unicode.add(pgpid)
+                block_label = unicodedata.normalize("NFC", block_label)
+            # add label to annotation
+            anno.body[0].label = block_label
+
+            # order according to block number
+            anno["schema:position"] = i + 1
+
+            # create database annotation
+            db_anno = Annotation()
+            db_anno.set_content(dict(anno))
+            # link to digital translation footnote
+            db_anno.footnote = footnote
+            db_anno.save()
+            # log entry to document annotation creation
+            self.log_addition(db_anno, "Migrated from Google Doc translation")
+
+            self.stats["created"] += 1
+
+        # export html/txt translation files to github backup
+        self.anno_exporter.export(
+            pgpids=[doc.pk],
+            commit_msg="Translation migrated from Google Doc - PGPID %d" % doc.pk,
```

Comment:
I would call this an import, not a migration

Reply:
wondering if you could set the base commit message on the class and then use it as the starting point for this one (reduce redundancy and also the unused commit message)